tests/chainer_tests/functions_tests/normalization_tests/test_standardize.py

```diff
@@ -0,0 +1,111 @@
+import functools
+import unittest
+
+import numpy
+
+from chainer.functions.normalization._standardize import _standardize
+from chainer import testing
+
+
+def _skip_if(cond, reason):
+    """Skip test if cond(self) is True"""
+    def decorator(impl):
+        @functools.wraps(impl)
+        def wrapper(self, *args, **kwargs):
+            if cond(self):
+                raise unittest.SkipTest(reason)
+            else:
+                impl(self, *args, **kwargs)
+        return wrapper
+    return decorator
+
+
+def _is_good_param(param):
+    # Check if 'nonzero' param is valid and meaningful. On the latter point,
+    # x should contain at least a zero if 'nonzeros' param is given.
+    return param['same'] == 'no'
+
+
+@testing.parameterize(*filter(_is_good_param, testing.product([
+    [
+        {'ch_out': 1},
+        {'ch_out': 5},
+    ],
+    [
+        {'size': 10},
+        {'size': 20},
+    ],
+    [
+        {'dtype': numpy.float32},
+        {'dtype': numpy.float16},
+    ],
+    [
+        # same (str): flag whether input elems are same values.
+        #   'no'   : all elems are randamly-chosen,
+        #   'equal': all elems are equal,
+        #   'near' : all elems are (randomly-chosen small values
+        #            + same value).
+        {'eps': 1e-5, 'same': 'no'},
+        {'eps': 1e-1, 'same': 'no'},
+        {'eps': 1e-1, 'same': 'equal'},
+        {'eps': 1e-1, 'same': 'near'},
+    ],
+])))
+@testing.backend.inject_backend_tests(
+    None,
+    # CPU tests
+    testing.product({
+        'use_cuda': [False],
+        'use_ideep': ['never', 'always'],
+    })
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
+    })
+    # ChainerX tests
+    + testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0'],
+    })
+)
+class TestStandardize(testing.FunctionTestCase):
+
+    def setUp(self):
+        self.skip_double_backward_test = self.same in ('equal', 'near')
+        if self.same == 'equal':
+            # Make self.x have same values
+            self.x[...] = self.x[0]
```

Comment:
I guess this processing should be done in `generate_inputs` method
and this test currently uses inputs generated randomly.